### PR TITLE
feat: disconnect button hides after hover out, click away, and improv…

### PR DESCRIPTION
**Pull request by natew**
disconnect button hides after hover out, click away, and improved animations and colors.

## Description

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2350/disconnect-button-is-not-dismissible

### Before

No hover leave or click away close + the animation wasn't smooth / setting icon doesn't animate.

### After

Show it once normal speed and then once slowed down

https://github.com/Uniswap/interface/assets/12100/d29863f4-b3f5-41fe-a588-3f1cf970b1a3

### Reproducing the error

1. 

### QA (ie manual testing)

- [ ] N/A

### Automated testing

- [ ] Unit test
- [ ] Integration/E2E test

### CODE REVIEW

File: src/components/AccountDrawer/IconButton.tsx [hunk 1]
```diff
@@ -136,41 +142,60 @@ export const IconWithConfirmTextButton = ({
...
     const keyHandler = (e: KeyboardEvent) => {
       if (e.key === 'Escape') {
-        setShowText(false)
-        e.preventDefault()
-        e.stopPropagation()
+        closeAndPrevent(e)
       }
     }
+
+    window.addEventListener('click', clickHandler, { capture: true })
```
Comment (just-toby) [inline src/components/AccountDrawer/IconButton.tsx:180]: just checking, did you try the existing `useOnClickOutside` hook and find that it didn't work for this case?

File: src/components/AccountDrawer/IconButton.tsx [hunk 2]
```diff
@@ -187,6 +212,18 @@ export const IconWithConfirmTextButton = ({
           setShowText(!showText)
         }
       }}
+      {...(dismissOnHoverOut && {
```
Comment (just-toby) [inline src/components/AccountDrawer/IconButton.tsx:215]: first time i've seen this method of optionally passing props. i like it

File: src/components/AccountDrawer/IconButton.tsx [hunk 3]
```diff
@@ -187,6 +212,18 @@ export const IconWithConfirmTextButton = ({
...
+        onMouseLeave() {
+          mouseLeaveTimeout.current = setTimeout(() => {
+            setShowText(false)
+          }, 500)
```
Comment (just-toby) [inline src/components/AccountDrawer/IconButton.tsx:219]: maybe give this value a name? `dismissOnHoverDurationMs` ?

**Review by just-toby — APPROVED** [review]
the animation looks good!

File: src/components/AccountDrawer/IconButton.tsx (hunk 1 above)
Comment (natew) [reply src/components/AccountDrawer/IconButton.tsx:180]: oh didn't know we had that

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by zzmp — APPROVED** [review]

**Pull request merged**